src/test/java/spoon/test/architecture/SpoonArchitectureEnforcerTest.java

```diff
@@ -212,6 +216,34 @@ public boolean matches(CtClass element) {
 	}
 
 	@Test
+	public void testInterfacesAreCtScannable() {
+		// contract: all non-leaf interfaces of the metamodel should be visited by CtInheritanceScanner
+		Launcher interfaces = new Launcher();
+		interfaces.addInputResource("src/main/java/spoon/support");
+		interfaces.addInputResource("src/main/java/spoon/reflect/declaration");
+		interfaces.addInputResource("src/main/java/spoon/reflect/code");
+		interfaces.addInputResource("src/main/java/spoon/reflect/reference");
+		interfaces.addInputResource("src/main/java/spoon/reflect/visitor/CtScanner.java");
+		interfaces.buildModel();
```

Comment:
SpoonMetaModel needs these packages too
				"spoon/support/reflect/declaration",
				"spoon/support/reflect/code",
				"spoon/support/reflect/reference"
Otherwise meta model is not complete.